test/plugin-barcode-scan.js

```diff
@@ -0,0 +1,58 @@
+var assert = require('assert'),
+    sinon = require('sinon'),
+    Gateway = require('../src/slave_gateway'),
+    BarcodeScan = require('../plugin-barcode-scan'),
+    barcodeScanPplugin = new BarcodeScan();
+
+
+describe('Barcode scan', function() {
+    var slaveInstance;
+
+    beforeEach(function () {
+        slaveInstance = Gateway();
+    });
```

Comment:
eslint complains about `beforeEach` not being recognized so `"mocha": true` should be included in`.eslintrc` `env` property